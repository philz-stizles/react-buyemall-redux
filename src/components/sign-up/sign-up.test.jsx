import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store';
import React from 'react'
import SignUpPage from './sign-up.component'

describe('Sign up Component', () => {
    it('should render SignUpPage with given state', () => {
        // const initialState = {
        //     menu: {
        //         items: [],
        //     }
        // };
    
        // const mockStore = configureStore()
        // const store = mockStore(initialState)
    
        expect(shallow(<SignUpPage />)).toMatchSnapshot()
    })

    it('correctly set state onchange', () => {
        // const wrapper = shallow(<SignUpPage />)
        // wrapper.find('[id="displayName"]').simulate('change', { target: { name: 'displayName', value: 'Theo' } })
        // console.log(wrapper.state())

        // const setFormState = jest.fn();
        const wrapper = mount(<SignUpPage />);
        
        // const handleChange = jest.spyOn(React, "useState");
        // handleChange.mockImplementation(formState => [formState, setFormState]);

        // wrapper.find("#displayName").simulate('change', { target: { name: 'displayName', value: 'Theo' } })
        // expect(setFormState).toEqual({
        //     displayName: 'Theo',
        //     email: '',
        //     password: '',
        //     confirmPassword: ''
        // })
    })
})