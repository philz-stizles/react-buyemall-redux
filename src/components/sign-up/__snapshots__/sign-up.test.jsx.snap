// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Component correctly set state onchange 1`] = `undefined`;

exports[`Sign up Component should render SignUpPage with given state 1`] = `ShallowWrapper {}`;

exports[`Sign up Component should render SignUpPage with given state 2`] = `ShallowWrapper {}`;
